kong/router.lua

```diff
@@ -662,6 +675,7 @@ do
   local matchers = {
     [MATCH_RULES.HOST] = function(route_t, ctx)
       local host = route_t.hosts[ctx.hits.host or ctx.req_host]
+        or route_t.hosts[utils.split_port(ctx.hits.host or ctx.req_host)]
```

Comment:
This may cause a lot of extra invocations of `split_port()` (for each evaluated route). Could we avoid it somehow? E.g. by storing `host_no_port` in the `ctx` table and only call `split_port` when `ctx.hits.host` is non-nil. Or even by calling `split_port` for each wildcard host value during initialization (in `marshall_route()`).

Reply:
i tried to store `host_no_port`, but seems there's something about lifetime that escapes me.  in the end I resorted to simply memoize `split_port()` itself since it's a pure function